### PR TITLE
Add standardized recommended/fixable/suggestions notices to rule docs

### PR DESCRIPTION
**Pull request by bmish**
Included changes:
* Adds standardized notices to the top of each rule doc to indicate whether the rule is in the `recommended` config, autofixable, or has automated suggestions
* Adds a test to verify each rule doc has the correct notices

These notices match the notices used by ESLint itself and many other ESLint plugins:

* https://eslint.org/docs/rules/no-useless-escape
* https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-get.md
* https://github.com/platinumazure/eslint-plugin-qunit/blob/master/docs/rules/no-assert-equal.md
* https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/master/docs/rules/no-only-tests.md
* https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-negated-condition.md

These notices ensure each rule doc clearly provides several critical data points by itself, without the reader having to refer to the [readme rules table](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/readme.md#rules) which has the same information.

### CODE REVIEW

File: docs/rules/catch-error-name.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Enforce a specific parameter name in catch clauses
 
+✅ The `"extends": "plugin:unicorn/recommended"` property in a configuration file enables this rule.
```
[the remark this answers is not in the document]
Comment (bmish) [reply docs/rules/catch-error-name.md]: Updated.

File: docs/rules/catch-error-name.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
...
+
+🔧 The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.
```
[the remark this answers is not in the document]
Comment (bmish) [reply docs/rules/catch-error-name.md]: Updated.

**Review by sindresorhus — APPROVED** [review]

**Comment by sindresorhus** [comment]
How should we handle cases like this? https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1586/files#diff-c6117e5638d17c7f095600f9b360e53d3f652539ca0d37d670d6bc4e50840cbdR5

**Comment by bmish** [comment]
> How should we handle cases like this? [#1586 (files)](https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1586/files#diff-c6117e5638d17c7f095600f9b360e53d3f652539ca0d37d670d6bc4e50840cbdR5)

I think it's fine for rule docs to optionally provide more detailed information about which specific cases are fixable or suggestable. In fact, some rule docs already include this information (`This rule is fixable when ...`), and I left this information in tact (in addition to my new standardized notices which are based purely on the rule meta properties).

**Review by fisker — APPROVED** [review]

**Pull request merged**